### PR TITLE
bug fix update

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - *Check output shape after op.is_dynamic()*
 - *Fix zero dimension allocation, set {1} dims to zero dimension shape*
 - *Add debug log for condition operation*

### Tickets:
 - *116928*

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/ops/constant.cpp [hunk 1]
```diff
@@ -97,8 +97,14 @@ static void create_data(ProgramBuilder& p, const ov::Shape& constDims, const std
         p.primitive_ids[initialconstPrimID] = constPrimID;
         p.profiling_ids.push_back(initialconstPrimID);
     } else {
-        GPU_DEBUG_LOG << "[" << initialconstPrimID << ": constant]" << std::endl;
+        if (constLayout.is_static() && constLayout.count() == 0) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/constant.cpp:100]: I think constLayout should always static ?

File: src/plugins/intel_gpu/src/plugin/ops/split.cpp [hunk 2]
```diff
@@ -27,6 +27,13 @@ static void CreateCommonSplitOp(ProgramBuilder& p, const std::shared_ptr<ov::Nod
             ov::Shape start_offset(input_pshape.size());
             for (size_t i = 0; i < op->get_output_size(); i++) {
                 const auto outPartialShape = op->get_output_partial_shape(i);
+                // op->is_dynamic() does not check if output shape is dynamic. it only check dynamism for input shapes
+                // Even if op->is_dynamic() is false, output shape can be dynamic.
+                // Thus, it is necessary to check if output shape is dynamic.
+                if (outPartialShape.is_dynamic()) {
+                    offsets.clear();
+                    break;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/split.cpp:35]: Shoundn't we prevent to go into the loop (at line28) at all and make it to proceed to the dynamic pass?

File: src/plugins/intel_gpu/src/plugin/ops/constant.cpp (hunk 1 above)
Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/plugin/ops/constant.cpp]: Fixed it. thanks

File: src/plugins/intel_gpu/src/plugin/ops/split.cpp (hunk 2 above)
Comment (ahnyoung-paul) [reply src/plugins/intel_gpu/src/plugin/ops/split.cpp]: Fixed it. thanks.

**Review by yeonbok — APPROVED** [review]

**Pull request merged**